procedures/diagnostics.sql

```diff
@@ -0,0 +1,1168 @@
+    DECLARE v_output_count, v_count, v_old_group_concat_max_len INT UNSIGNED DEFAULT 0;
+    -- The width of each of the status outputs in the summery
+    DECLARE v_status_summary_width TINYINT UNSIGNED DEFAULT 50;
+    DECLARE v_done BOOLEAN DEFAULT FALSE;
+    -- Do not include the following ndbinfo views:
+    --    'blocks'                    Static
+    --    'config_params'             Static
+    --    'dict_obj_types'            Static
+    --    'disk_write_speed_base'     Can generate lots of output - only include aggregate views here
+    --    'memory_per_fragment'       Can generate lots of output
+    --    'memoryusage'               Handled separately
+    --    'operations_per_fragment'   Can generate lots of output
+    --    'threadblocks'              Only needed once
+    DECLARE c_ndbinfo CURSOR FOR
+        SELECT TABLE_NAME
+          FROM information_schema.TABLES
+         WHERE TABLE_SCHEMA = 'ndbinfo'
+               AND TABLE_NAME NOT IN (
+                  'blocks',
+                  'config_params',
+                  'dict_obj_types',
+                  'disk_write_speed_base',
+                  'memory_per_fragment',
+                  'memoryusage',
+                  'operations_per_fragment',
+                  'threadblocks'
+               );
+    DECLARE c_sysviews_w_delta CURSOR FOR
+        SELECT table_name
+          FROM tmp_sys_views_delta
+         ORDER BY table_name; 
+    DECLARE CONTINUE HANDLER FOR NOT FOUND SET v_done = TRUE;
+
+    -- Some metrics variables doesn't make sense in delta and rate calculations even if they are numberic
+    -- as they really are more like settings or "current" status.
+    SET v_no_delta_names = CONCAT('s%{COUNT}.Variable_name NOT IN (',
+        '''innodb_buffer_pool_pages_total'', ',
+        '''innodb_page_size'', ',
+        '''last_query_cost'', ',
+        '''last_query_partial_plans'', ',
+        '''qcache_total_blocks'', ',
+        '''slave_last_heartbeat'', ',
+        '''ssl_ctx_verify_depth'', ',
+        '''ssl_ctx_verify_mode'', ',
+        '''ssl_session_cache_size'', ',
+        '''ssl_verify_depth'', ',
+        '''ssl_verify_mode'', ',
+        '''ssl_version'', ',
+        '''buffer_flush_lsn_avg_rate'', ',
+        '''buffer_flush_pct_for_dirty'', ',
+        '''buffer_flush_pct_for_lsn'', ',
+        '''buffer_pool_pages_total'', ',
+        '''lock_row_lock_time_avg'', ',
+        '''lock_row_lock_time_max'', ',
+        '''innodb_page_size''',
+    ')');
+
+    -- Do not track the current thread - no reason to clutter the output
+    SELECT INSTRUMENTED INTO v_this_thread_enabled FROM performance_schema.threads WHERE PROCESSLIST_ID = CONNECTION_ID();
+    CALL sys.ps_setup_disable_thread(CONNECTION_ID());
+
+    -- Set configuration options
+    IF (@sys.allow_i_s_tables IS NULL) THEN
+        SET @sys.allow_i_s_tables          = sys.sys_get_config('allow_i_s_tables'         , 'OFF');
```

Comment:
This variable, and the include_raw and debug ones, should also be included in the sys.sys_config table, right? I don't see those added here.. I would prefer allow_i_s_tables to also be prefixed with "diagnostics" consistently with "diagnostics.include_raw" too (I like the idea of prefixing with routine names for routine specific options).